services/packagist/packagist-downloads.service.js

```diff
@@ -1,28 +1,58 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
 const { metric } = require('../text-formatters')
-const { downloadCount: downloadCountColor } = require('../color-formatters')
-
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class PackagistDownloads extends LegacyService {
-  static get category() {
-    return 'downloads'
-  }
+const { downloadCount } = require('../color-formatters')
+const { keywords, BasePackagistService } = require('./packagist-base')
 
+module.exports = class PackagistDownloads extends BasePackagistService {
   static get route() {
     return {
       base: 'packagist',
       pattern: ':interval(dm|dd|dt)/:user/:repo',
     }
   }
 
+  static get defaultBadgeData() {
+    return {
+      label: 'downloads',
+    }
+  }
+
+  async handle({ interval, user, repo }) {
+    const {
+      package: { downloads },
+    } = await this.fetch({ user, repo })
+
+    return this.constructor.render({ downloads, interval })
+  }
+
+  static render({ downloads, interval }) {
+    let amount
+    let ending = ''
+
+    switch (interval) {
```

Comment:
This feels a bit like the render function is also performing the transform/data parsing. 

A common approach used with other badges that leverage is intervals is an object map (such as [pypi downloads](https://github.com/badges/shields/blob/master/services/pypi/pypi-downloads.service.js#L19)).

With something like that the render function could simply take in the download number, the responsibility to extract of the correct download number from the response could shift back to the handle function, the examples would simplify, etc.